wherehows-web/app/components/dataset-aclaccess.ts

```diff
@@ -0,0 +1,141 @@
+import Component from '@ember/component';
+import ComputedProperty from '@ember/object/computed';
+import { inject } from '@ember/service';
+import { computed, getProperties, get, set } from '@ember/object';
+import { task } from 'ember-concurrency';
+
+import { getAclAccess } from 'wherehows-web/utils/api/datasets/acl-access';
+import { accessState, pickList, getPrincipal } from 'wherehows-web/constants/dataset-aclaccess';
+import Notifications, { NotificationEvent } from 'wherehows-web/services/notifications';
+import { IAclInfo, IRequestAclReject, IRequestAclApproved } from 'wherehows-web/typings/api/datasets/aclaccess';
+
+export default class DatsetAclAccess extends Component {
+  constructor() {
+    super(...arguments);
+    this.resetForm = this.resetForm.bind(this);
+    this.actions.cancelTasks = this.actions.cancelTasks.bind(this);
+    this.actions.addOwner = this.actions.addOwner.bind(this);
+  }
+  /**
+   * Reference to the application notifications Service
+   * @type {ComputedProperty<Notifications>}
+   * @memberOf DatsetAclAccess
+   */
+  notifications = <ComputedProperty<Notifications>>inject();
+
+  /**
+   * Define property binds to the textarea
+   * @type {string}
+   * @memberOf DatsetAclAccess
+   */
+  requestReason: string;
+
+  /**
+   * Define the property to show the user info
+   * @type {string}
+   * @memberOf DatsetAclAccess
+   */
+  currentUser: string;
+
+  /**
+   * Define the property to initialize the page 
+   * @type {IAclInfo}
+   * @memberOf DatsetAclAccess
+   */
+  accessInfo: IAclInfo;
+
+  /**
+   *  Define the property to update the page when a user is granted permission.
+   * @type {IRequestAclReject | IRequestAclApproved}
+   * @memberOf DatsetAclAccess
+   */
+  accessResponse: IRequestAclReject | IRequestAclApproved;
+
+  /**
+   * Define the computed property to decide the page state. 
+   * The component has 5 states ['emptyState', 'hasAccess','noAccess','denyAccess','getAccess'].
+   * @type {string}
+   * @memberOf DatsetAclAccess
+   */
+  pageState: ComputedProperty<string> = computed('accessInfo', 'accessResponse', function(this: DatsetAclAccess) {
```

Comment:
I guess this is ok atm. However, I think using tasks would be a better approach here, rather than creating intermediate state like `isLoadPage`, you can use the derived state from a task

Reply:
Also, these states could probably be stored in an externally defined enum, rather than being internal state